src/main/scala/org/constellation/infrastructure/p2p/client/SignClientInterpreter.scala

```diff
@@ -6,31 +6,32 @@ import org.constellation.infrastructure.p2p.PeerResponse.PeerResponse
 import org.http4s.client.Client
 import org.constellation.domain.p2p.client.SignClientAlgebra
 import org.constellation.p2p.{PeerAuthSignRequest, PeerRegistrationRequest}
+import org.constellation.session.SessionTokenService
 import org.constellation.util.SingleHashSignature
 import org.http4s.circe.CirceEntityDecoder._
 import org.http4s.circe.CirceEntityEncoder._
 import org.http4s.Method._
 
-class SignClientInterpreter[F[_]: ContextShift](client: Client[F])(implicit F: Concurrent[F])
+class SignClientInterpreter[F[_]: ContextShift](client: Client[F], sessionTokenService: SessionTokenService[F])(implicit F: Concurrent[F])
     extends SignClientAlgebra[F] {
 
   def sign(authSignRequest: PeerAuthSignRequest): PeerResponse[F, SingleHashSignature] =
-    PeerResponse("sign", client, POST) { (req, c) =>
+    PeerResponse("sign", client, POST, sessionTokenService, verifyResponseToken = false) { (req, c) =>
```

Comment:
hm, it's kinda red light for my, that I need to pass session token service but actually I don't want to use it. can we make it dependant on the verifyResponseToken?

Reply:
Good point. I will try to think of something.